python/GafferUI/RampPlugValueWidget.py

```diff
@@ -78,35 +78,34 @@ def __init__( self, plug, **kw ) :
 			self.__lastPositionChangedReason = None
 			self.__positionsMergeGroupId = 0
 
-			with GafferUI.ListContainer(
-				orientation = GafferUI.ListContainer.Orientation.Horizontal if isinstance( plug.pointYPlug( 0 ), Gaffer.FloatPlug ) else GafferUI.ListContainer.Orientation.Vertical,
-				spacing = 4
-			) :
-
-				self.__positionLabel = GafferUI.LabelPlugValueWidget(
-					plug.pointXPlug( 0 ),
-					parenting = { "verticalAlignment" : GafferUI.VerticalAlignment.Top }
-				)
-				self.__positionField = GafferUI.NumericPlugValueWidget(
-					plug.pointXPlug( 0 ),
-					parenting = { "verticalAlignment" : GafferUI.VerticalAlignment.Top }
-				)
-
-				self.__valueLabel = GafferUI.LabelPlugValueWidget(
-					plug.pointYPlug( 0 ),
-					parenting = { "verticalAlignment" : GafferUI.VerticalAlignment.Top }
-				)
-				if isinstance( plug.pointYPlug( 0 ), Gaffer.FloatPlug ):
-					self.__valueField = GafferUI.NumericPlugValueWidget(
-						plug.pointYPlug( 0 ),
-						parenting = { "verticalAlignment" : GafferUI.VerticalAlignment.Top }
-					)
-				else:
-					self.__valueField = GafferUI.ColorPlugValueWidget(
-						plug.pointYPlug( 0 ),
-						parenting = { "verticalAlignment" : GafferUI.VerticalAlignment.Top }
-					)
-					self.__valueField.setColorChooserVisible( True )
+			self.__positionLabel = GafferUI.LabelPlugValueWidget( plug.pointXPlug( 0 ) )
+			self.__positionField = GafferUI.NumericPlugValueWidget( plug.pointXPlug( 0 ) )
+
+			self.__valueLabel = GafferUI.LabelPlugValueWidget( plug.pointYPlug( 0 ) )
+			if isinstance( plug.pointYPlug( 0 ), Gaffer.FloatPlug ) :
+				self.__valueField = GafferUI.NumericPlugValueWidget( plug.pointYPlug( 0 ) )
+			else:
+				self.__valueField = GafferUI.ColorPlugValueWidget( plug.pointYPlug( 0 ) )
+				self.__valueField.setColorChooserVisible( True )
+
+			if isinstance( plug.pointYPlug( 0 ), Gaffer.FloatPlug ) :
+				row = GafferUI.ListContainer( orientation = GafferUI.ListContainer.Orientation.Horizontal, spacing = 4 )
+
+				row.addChild( self.__positionLabel, verticalAlignment = GafferUI.VerticalAlignment.Top )
+				row.addChild( self.__positionField, verticalAlignment = GafferUI.VerticalAlignment.Top )
+
+				row.addChild( self.__valueLabel, verticalAlignment = GafferUI.VerticalAlignment.Top )
+				row.addChild( self.__valueField, verticalAlignment = GafferUI.VerticalAlignment.Top )
+			else :
+				grid = GafferUI.GridContainer( spacing = 4 )
```

Comment:
Maybe you could use a single `with GridContainer()` for both cases, and then the only thing that would differ for the float and color modes would be the indices of the widgets within the grid?

Reply:
That was my initial approach, but I wasn't super happy with the layout for float splines. Here's what that looks like :
![image](https://github.com/GafferHQ/gaffer/assets/7873467/822ebe33-036f-428d-8937-db6d756a2946)

It seems odd to me that it doesn't fill the width, but maybe that's just me being used to the old style. Maybe there's a way to expand the widgets, `setSizePolicy()` sounds like a good place to try.